src/main/groovy/cruise/umple/UmpleGradlePlugin.groovy

```diff
@@ -29,12 +29,14 @@ class UmpleGradlePlugin implements Plugin<Project> {
 		project.task('generateSource') << {		
 			// command line arguments can be specified through gradle by -P (-P is for project properties)
 			// eg: "gradle compileUmpleFile -PUMPLE_FILE_PATH=test.ump -PLANGUAGE_TO_GENERATE=Java"	
+			m_umpleFilePath = "${project.projectDir}${File.separator}" 
+			m_generatedOutputPath = "${project.projectDir}${File.separator}"
 		
 			if(project.hasProperty(UMPLE_FILE_PATH))
 			{
-				m_umpleFilePath = project.getProperty(UMPLE_FILE_PATH)	
+				m_umpleFilePath += project.getProperty(UMPLE_FILE_PATH)	
```

Comment:
> I'm not sure what to make of that. Are you trying to tell me to ask fewer questions? I'm happy to take more ownership of the project, you seem to have a specific vision for how this project will turn out, though. I'm just trying to make sure we're both on the same page.

Nooooo, oh God. I was.. going to the bathroom during a movie and I guess my connection crapped out. Oops. Didn't mean to send twice. 

> The paths that the user specifies, as well as the default path constants, are relative to the project's main directory (i.e. the folder that contains build.gradle). If the project structure changes then the default values need to be changed to reflect the change.

As long as it works, that's what matters! If you can verify that, I'm happy.